services-core/src/economic_viability.rs

```diff
@@ -31,29 +32,31 @@ impl EconomicViabilityStrategy {
         native_token_price: Arc<dyn NativeTokenPricing + Send + Sync>,
         gas_station: Arc<dyn GasPriceEstimating + Send + Sync>,
     ) -> Result<Arc<dyn EconomicViabilityComputing>> {
+        let make_dynamic = || {
```

Comment:
The reason for making this a helper closure is that in the match statement below it avoids code duplication.